### PR TITLE
APP-13004 Header Navigation

### PR DESCRIPTION
**Pull request by webspaceadam**
- [Ticket](https://flipapp.myjetbrains.com/youtrack/issue/APP-13004/Header-Navigation-Design-Improvements)

### CODE REVIEW

File: apps/swirl-docs/src/components/HeaderNavigation/index.tsx [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
 
 const HeaderNavigation = () => {
-  const [isMobileNavOpen, setIsMobileNavOpen] = useState(false);
   const router = useRouter();
+  const [isMobileNavOpen, setIsMobileNavOpen] = useState(false);
+
+  useEffect(() => {
+    if (typeof window) {
+      if (isMobileNavOpen) {
+        document.getElementsByTagName("body")[0].style.overflow = "hidden";
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/HeaderNavigation/index.tsx:34]: **suggestion:** You could consider using https://github.com/willmcpo/body-scroll-lock here. It solves all the edge cases on mobile devices.

File: apps/swirl-docs/src/components/HeaderNavigation/index.tsx [hunk 2]
```diff
@@ -76,21 +76,51 @@ const HeaderNavigation = () => {
...
+      </DesktopView>
+      <MobileView>
+        <header className="sticky top-0 flex flex-col w-full h-full max-h-screen bg-background-default">
+          <div
+            aria-label="main"
+            className="flex justify-between items-center h-16 w-full px-4 border-b-1 font-normal text-base"
+          >
+            <HeaderLogo />
+            {isMobileNavOpen ? (
+              <button type="button" onClick={() => setIsMobileNavOpen(false)}>
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/HeaderNavigation/index.tsx:102]: **suggestion:** For better accessibility, the toggle should be inside the navigations `nav` tags (or set `aria-controls` on the button, if not possible). You could also set `aria-expanded="true|false"`. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#menu

File: apps/swirl-docs/src/components/HeaderNavigation/index.tsx [hunk 3]
```diff
@@ -76,21 +76,51 @@ const HeaderNavigation = () => {
...
+                <Image alt="Close Menu" src={icon.src} width={24} height={24} />
+              </button>
+            ) : (
+              <button
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/HeaderNavigation/index.tsx:106]: **suggestion:** see above

File: apps/swirl-docs/src/components/HeaderNavigation/index.tsx (hunk 1 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/components/HeaderNavigation/index.tsx:34]: Had no Idea a lib for that existed. I will take a look into it and implement it :)

**Review by Sqrrl — APPROVED** [review]

**Pull request merged**